### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by ben-edmondson**

### CODE REVIEW

File: BowlingKata/run.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿/*
```
Comment (AlexBeesley) [inline BowlingKata/run.cs:1]: This should be stilled called `Program.cs`

File: BowlingKata/Rolling.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Formats.Asn1.AsnWriter;
+
+namespace Bowling
+{
+    public class Rolling
+    {
+
```
Comment (AlexBeesley) [inline BowlingKata/Rolling.cs:12]: Try go through and delete any empty space like this, its good to have blank lines within methods to separate different logic, but you should not have random blank lines in places like this.

File: BowlingKata/Rolling.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        public static List<int> RollAll()
+        {
+            List<int> score = new List<int>();
+
+            //loops 10 times and gives a number
```
Comment (AlexBeesley) [inline BowlingKata/Rolling.cs:17]: comments are taught at uni, but in practice they aren't used much in industry, especially at TPR, your code should be readable through your choice of variable names and its structure.

File: BowlingKata/Scoring.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        public static int ScoreFinal(List<int> scores)
+        {
+            List<int> score = scores;
+            //tallies scores including bonuses for strikes/spares
+            int scoreOne = 0;
+            int scoreTwo = 1;
+            int finalScore = 0;
+            for (int i = 0; i < 10; i++)
+            {
+                switch (score[scoreOne])
```
Comment (AlexBeesley) [inline BowlingKata/Scoring.cs:20]: Switch case statements are useful but in this case an IF statement would be more appropriate.

if you wish to understand this a bit more I suggest having a read through this:
https://stackoverflow.com/questions/395618/is-there-any-significant-difference-between-using-if-else-and-switch-case-in-c

File: BowlingKata/Scoring.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            List<int> scoreBoard = finalScores;
+            Console.WriteLine("Your final score will now be calculated..");
+
+            int scoreOne = 0;
+            int scoreTwo = 1;
+            for (int i = 0; i < 10; i++)
+            {
+                if (scoreBoard[scoreOne] == 10)
+                {
+                    Console.Write("X");
```
Comment (AlexBeesley) [inline BowlingKata/Scoring.cs:65]: All your user inputs and outputs should be handled in the `Program.cs` class, write this method in a way which only returns the score values, do all `Console.Write()` lines in the `Program.cs` file.

File: BowlingKata/run.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+        static void Main(string[] args)
+        {
+            //menu + loop
+            bool gameLoop = true;
+            string restartGame;
+            while (gameLoop == true)
+            {
+                //Rolls and does scoreboard
+                List<int> score = Rolling.BonusScoresAdded(Rolling.RollAll());
```
Comment (AlexBeesley) [inline BowlingKata/run.cs:26]: Look into the other `Console.` extensions, you can clear everything and close the app with that.

File: BowlingTests/RollingTest.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+using Bowling;
+
+namespace BowlingTests
+{
+    public class RollingTest
+    {
+        //test types and if null
+        [Test]
```
Comment (AlexBeesley) [inline BowlingTests/RollingTest.cs:8]: As discussed, add some tests here, even simple ones to check the return is of a certain type and not null.

File: BowlingTests/ScoringTest.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            // Arrange
+            List<int> rolls = new List<int> { 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5 };
+            var result = 150;
+            // Act
+            var actual = Scoring.ScoreFinal(rolls);
+
+            // Assert
+            Assert.That(result, Is.EqualTo(actual));
+        }
+    }
```
Comment (AlexBeesley) [inline BowlingTests/ScoringTest.cs:70]: Well done here, these tests cover all your bases. Consider what might happen if a method returns null, test for this and change your code to handle that exception with breaking.

File: BowlingKata/run.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    class Run
+    {
+
+
+        static void Main(string[] args)
+        {
+            //menu + loop
+            bool gameLoop = true;
+            string restartGame;
```
Comment (AlexBeesley) [inline BowlingKata/run.cs:22]: Make this nullable to get rid of the `CS8600	Converting null literal or possible null value to non-nullable type.` warning.

**Review by AlexBeesley — CHANGES_REQUESTED** [review]
Have had a look and provided some feedback. Overall, great work.
